listenbrainz/webserver/static/js/src/stats/UserHistory.tsx

```diff
@@ -256,27 +210,85 @@ export default class UserHistory extends React.Component<
     return result;
   };
 
-  handleError = (error: Error): void => {
-    // Error Boundaries don't catch errors in async code.
-    // Throwing an error in setState fixes this.
-    // This is a hacky solution but should be fixed with upcoming concurrent mode in React.
-    this.setState(() => {
-      throw error;
-    });
+  syncStateWithURL = async (): Promise<void> => {
+    try {
+      const { page, range, entity } = this.getURLParams();
+      const { range: currRange, entity: currEntity } = this.state;
+      if (range !== currRange || entity !== currEntity) {
+        const {
+          totalPages,
+          maxListens,
+          entityCount,
+          startDate,
+        } = await this.getInitData(range, entity);
+        this.setState({
+          range,
+          entity,
+          startDate,
+          totalPages,
+          maxListens,
+          entityCount,
+        });
+      }
+      const data = await this.getData(page, range, entity);
+      this.setState({
+        data: this.processData(data, page, entity),
+        currPage: page,
+      });
```

Comment:
I think the weird transition jump that @paramsingh mentioned (https://github.com/metabrainz/listenbrainz-server/pull/900#pullrequestreview-425086819) might be caused by how setState is being called in succession here, considering setState is asynchronous.

I would suggest doing a single call, maybe something like this (pseudocode warning):
```
let initData = {};
if (range !== currRange || entity !== currEntity) {
  initData = await this.getInitData(range, entity);
}
const data = await this.getData(page, range, entity);
this.setState({
  data: this.processData(data, page, entity),
  currPage: page,
  ...initData
});
```

Let me know if that works and fixes the issue.

Reply:
Yep, that fixed it, thanks for the tip :smiley: